packages/opentelemetry-metrics/src/Metric.ts

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { hashLabelValues } from './Utils';
+import { CounterHandle, GaugeHandle } from './Handle';
+import { MetricOptions } from './types';
+
+/** This is a SDK implementation of {@link Metric} interface. */
+export abstract class Metric<T> implements types.Metric<T> {
+  protected readonly _monotonic: boolean;
+  protected readonly _disabled: boolean;
+  private readonly _handles: Map<String, T> = new Map();
+
+  constructor(name: string, options: MetricOptions) {
+    this._monotonic = options.monotonic;
+    this._disabled = options.disabled;
+  }
+
+  /**
+   * Returns a Handle associated with specified label values.
+   * It is recommended to keep a reference to the Handle instead of always
+   * calling this method for each operation.
+   * @param labelValues the list of label values.
+   */
+  getHandle(labelValues: string[]): T {
```

Comment:
I think getHandle isn't meant to take `labelValues`, but an instance of a `LabelSet` and just be a wrapper around a specific metric.

```typescript
const labelSet1: LabelSet = meter.defineLabels(labels1)
const labelSet2: LabelSet = meter.defineLabels(labels2)

const counter = meter.createCounter("name");

counter.add(labelSet1, 1); // counter now has 1 for labelSet1

const handle = counter.getHandle(labelSet2);
handle.add(1) // counter has 1 for labelSet2
```